lib/puppet/provider/package/pkgin.rb

```diff
@@ -30,7 +30,7 @@ def self.prefetch(packages)
   end
 
   def self.instances
-    pkgin(:list).split("\n").map do |package|
+    pkgin("-p", :list).split("\n").map do |package|
```

Comment:
-p was added in 0.7.0 which was released end of last year and is part of pkgsrc 2015Q1. There is no backwards compat for pkgsrc so it's a safe assumption that people using pkgin are running with at least 0.7.0. Otherwise the rest of their software isn't "supported" either.